### PR TITLE
Key view looping accessibility for macOS

### PR DESCRIPTION
**Pull request by HeyImChris**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [X] I am making a fix / change for the macOS implementation of react-native
- [X] I am making a change required for Microsoft usage of react-native

## Summary

Key view loop support. This is incomplete at the moment, but making a PR so it can be picked by anyone else. Remaining work is hooking up the test page I made to actually set non-sequential controls as their next and previous views to verify the APIs work as expected.

## Changelog

[Apple] [Feature] - Key-view loop accessibility

## Test Plan

TBD

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-macos/pull/746)

### CODE REVIEW

File: Libraries/Components/Button.js [hunk 1]
```diff
@@ -154,6 +155,40 @@ type ButtonProps = $ReadOnly<{|
    * For arrow keys, add "ArrowLeft", "ArrowRight", "ArrowUp", "ArrowDown",
    */
   validKeysUp?: ?Array<string>,
+  
+  /**
+   * Specifies whether the view can become key view
+   */
+  canBecomeKeyView?: ?boolean, // TODO(macOS ISS#2323203)
```
Comment (appden) [inline Libraries/Components/Button.js:162]: How will this property differ from the existing `focusable` prop?

File: Libraries/Components/Button.js [hunk 2]
```diff
@@ -154,6 +155,40 @@ type ButtonProps = $ReadOnly<{|
...
+  /**
+  * The view object that follows the current view in the key view loop.
+  */
+  nextKeyView?: ?RCTView, // TODO(macOS ISS#2323203)
+  
+  /**
+  * The closest view object in the key view loop that follows the current 
+  * view in the key view loop and accepts first responder status.
+  */
+  nextValidKeyView?: ?RCTView, // TODO(macOS ISS#2323203)
```
Comment (appden) [inline Libraries/Components/Button.js:179]: This is usually readonly on the native side. I'm not sure in which cases you'd want to set this separately from `nextKeyView`.

File: Libraries/Components/Button.js (hunk 1 above)
Comment (Saadnajmi) [reply Libraries/Components/Button.js:162]: Hm, maybe `focusable` should internally map to this property, or it's setter sets this property as well?

File: Libraries/Components/Button.js (hunk 2 above)
Comment (Saadnajmi) [reply Libraries/Components/Button.js:179]: Good catch, we might want to remove this. I'm investigating if we actually do need this for anything

Comment (Saadnajmi) [reply Libraries/Components/Button.js:179]: Oh yeah, also @appden nice to meet you! I work with Chris and I should be taking over this PR to get it checked in. Just an FYI =)

File: Libraries/Components/Button.js (hunk 1 above)
Comment (HeyImChris) [reply Libraries/Components/Button.js:162]: Probably seems fair to map focusable here

File: Libraries/Components/Button.js (hunk 2 above)
Comment (HeyImChris) [reply Libraries/Components/Button.js:179]: Good point- I think the readonly props should stay readonly

**Comment by HeyImChris** [comment]
Closing as we've got another review out that handles this

**Pull request closed without merging**